cmd/kogito/command/iozip/zip_files.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iozip
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/context"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/flag"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/message"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var supportedS2iExtensions = []string{".dmn", ".drl", ".bpmn", ".bpmn2", ".properties"}
+var supportedJvmExtensions = []string{".jar", ".json"}
+var supportedNativeExtensions = []string{"-runner", ".json"}
+
+func zipFile(absoluteFilePath string, fileInfo os.FileInfo, resource string, binaryBuildType flag.BinaryBuildType, tarWriter *tar.Writer) (string, error) {
```

Comment:
the package calls `iozip` and the function `zipFile`. Take care when naming packages to not have things like `iozip.Zip(...)`. In this case the function is private, but worth a heads up.

Reply:
What would be wrong with `iozip.Zip()`?

Reply:
I don't have the time to go through it now, but we have linters that check for package/function names. `zip.Zip` for instance. It's not a best practice. This was just a heads up since some linters can get you off guard.